### PR TITLE
782 part1

### PR DESCRIPTION
**Pull request by manikey123**
this is the part to increase the hard coded limit on the get_account_history_operations api call

### CODE REVIEW

File: libraries/app/api.cpp [hunk 1]
```diff
@@ -352,7 +352,8 @@ namespace graphene { namespace app {
     {
        FC_ASSERT( _app.chain_database() );
        const auto& db = *_app.chain_database();
-       FC_ASSERT( limit <= 100 );
+        uint32_t max_account_history_operations_limit=_app.get_options().max_account_history_operations_limit;
```
Comment (oxarbitrage) [inline libraries/app/api.cpp]: I think you can add `const` here. Also please use codebase indentation of 3 spaces.

File: libraries/app/application.cpp [hunk 2]
```diff
@@ -439,6 +439,13 @@ void application_impl::startup()
    if ( _options->count("enable-subscribe-to-all") )
       _app_options.enable_subscribe_to_all = _options->at( "enable-subscribe-to-all" ).as<bool>();
 
+   if ( _options->count("max-account-history-operations-limit") )
+   {
+      uint32_t max_acct_hist_limit = _options->at("max-account-history-operations-limit").as<uint32_t>();
+      _app_options.max_account_history_operations_limit = _options->at("max-account-history-operations-limit").as<uint32_t>();
```
Comment (oxarbitrage) [inline libraries/app/application.cpp]: you are doing the same thing twice, `max_acct_hist_limit` is not used.

File: libraries/app/application.cpp [hunk 3]
```diff
@@ -982,6 +989,8 @@ void application::set_program_options(boost::program_options::options_descriptio
          ("enable-standby-votes-tracking", bpo::value<bool>()->implicit_value(true),
           "Whether to enable tracking of votes of standby witnesses and committee members. "
           "Set it to true to provide accurate data to API clients, set to false for slightly better performance.")
+           ("max-account-history-operations-limit",boost::program_options::value<uint32_t>()->default_value(100),
```
Comment (oxarbitrage) [inline libraries/app/application.cpp]: indentation.

File: libraries/app/application.cpp (hunk 2 above)
Comment (oxarbitrage) [reply libraries/app/application.cpp]: default value of 100 should be added here. also, i am not sure if the option should work for ALL the hardcoded limits or only for the account history related. If it is for all hardcoded values inside the API then the name of the option is not very well picked.

File: libraries/app/include/graphene/app/application.hpp [hunk 4]
```diff
@@ -40,6 +40,7 @@ namespace graphene { namespace app {
       public:
          bool enable_subscribe_to_all = false;
          bool has_market_history_plugin = false;
+         uint32_t max_account_history_operations_limit=100;
```
Comment (oxarbitrage) [inline libraries/app/include/graphene/app/application.hpp]: I think this is not needed.

File: libraries/app/application.cpp (hunk 2 above)
Comment (manikey123) [reply libraries/app/application.cpp]: removed extra line at 444

File: libraries/app/include/graphene/app/application.hpp (hunk 4 above)
Comment (manikey123) [reply libraries/app/include/graphene/app/application.hpp]: -Not needed for base scenario 
-However needed for scenario from test case where initialization is not provided. Additionally this is part of the clarification I needed relating to below. More details in the main ticket (#782)

Needed clarification on :
1.
On running function get_account_history_operations located in file history_api_test.cpp, config.init file is not loaded , can you suggest any approach to test the API ?
image

I am using ilog function for debugging the shared_pointers as I face BOOST error for the RTTI symbols not found. In my company the team uses CLION IDE with GDB and found it to be more time-saving than ilog. I am open for any better way to debug and open to connect with any technical member for the same.

Comment (pmconrad) [reply libraries/app/include/graphene/app/application.hpp]: > On running function get_account_history_operations located in file history_api_test.cpp, config.init file is not loaded , can you suggest any approach to test the API ?

To my knowledge, we don't have tests that use config files. Best approach might be to derive a subclass from database_fixture that creates and uses a config file in `graphene::utilities::temp_directory_path()`.

Comment (manikey123) [reply libraries/app/include/graphene/app/application.hpp]: > > On running function get_account_history_operations located in file history_api_test.cpp, config.init file is not loaded , can you suggest any approach to test the API ?
> 
> To my knowledge, we don't have tests that use config files. Best approach might be to derive a subclass from database_fixture that creates and uses a config file in `graphene::utilities::temp_directory_path()`.

Thank you I shall explore above approach.
In addition if I start witness node and hit the API for function get_account_history_operations, would this option work? I shall explore this too and update.

**Pull request closed without merging**